crates/fuel-core/src/schema/node_info.rs

```diff
@@ -108,25 +106,26 @@ impl NodeQuery {
 
         let db = ctx.data_unchecked::<ReadDatabase>();
         let read_view = db.view()?;
-        let indexation = HashSet::from_iter(
-            [
-                (
-                    read_view.balances_indexation_enabled,
-                    IndexationKind::Balances,
-                ),
-                (
-                    read_view.coins_to_spend_indexation_enabled,
-                    IndexationKind::CoinsToSpend,
-                ),
-                (
-                    read_view.asset_metadata_indexation_enabled,
-                    IndexationKind::AssetMetadata,
-                ),
-            ]
-            .into_iter()
-            .filter_map(|(enabled, kind)| enabled.then_some(kind)),
-        );
-
+        let mut indexation = Indexation::new();
+        for kind in IndexationKind::iter() {
```

Comment:
Very good way to enforce compiler error=D

Reply:
:p https://github.com/FuelLabs/fuel-core/pull/2595#discussion_r1922922496

Reply:
a bit less optimised but worth the cost imo